### PR TITLE
#376: Update grammar to Swift 2.2

### LINKED ISSUE #376: Update grammar to support Swift 2.2

### PR DESCRIPTION
**Pull request by a4sriniv**
Changelog:

- Remove C-Style for loops
- Function signatures can only have a single parameter clause (no more curried functions)
- Class declarations are only allowed to have access level modifiers and 'final'
- Protocol associated types now use 'assoicatedtype' keyword
- Tuple pattern element syntax updated
- Initializer expressions can now contain "argument names"
- Language version testing function added to build configurations

Resolves #376.

### CODE REVIEW

File: src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4 [hunk 1]
```diff
@@ -345,8 +337,9 @@ structBody : '{' declarations?'}'  ;
 
 // GRAMMAR OF A CLASS DECLARATION
 
-// declarationModifier missing in Swift Language Reference
-classDeclaration : attributes? declarationModifier* 'class' className genericParameterClause? typeInheritanceClause? classBody  ;
+classDeclaration : attributes? classDeclarationModifiers? 'class' className genericParameterClause? typeInheritanceClause? classBody  ;
+// Swift Language Reference does not allow final to come before accessLevelModifier
+classDeclarationModifiers: accessLevelModifier 'final'? | 'final' accessLevelModifier? ;
```
Comment (adityatrivedi) [inline src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4]: Can this be rewritten as:
```ANTLR
classDeclarationModifiers: accessLevelModifier? 'final'? ;
```

File: src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4 [hunk 2]
```diff
@@ -597,10 +589,10 @@ dictionaryLiteralItem : expression ':' expression  ;
 
 selfExpression
  : 'self'
- | 'self' '.' identifier
+ | 'self' '.' identifier  // self-method-expression
  // Swift Language Reference uses expressionList
- | 'self' '[' expressionElementList ']'
- | 'self' '.' 'init'
+ | 'self' '[' expressionElementList ']'  // self-subscript-expression
+ | 'self' '.' 'init' // self-initializer-expression
```
Comment (adityatrivedi) [inline src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4]: Can you add a test for `.init` containing a `dynamicType`?

```swift
let s3 = someValue.dynamicType.init(data: 7)
```

**Comment by adityatrivedi** [comment]
Fantastic work! 🌟

File: src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4 (hunk 1 above)
Comment (a4sriniv) [inline src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4]: Nope, this wouldn't be parsed correctly: `final public class Car {}`

File: src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4 (hunk 2 above)
Comment (a4sriniv) [inline src/main/antlr/com/sleekbyte/tailor/antlr/Swift.g4]: Done.

**Comment by adityatrivedi** [comment]
:+1:

**Comment by alykhank** [comment]
:+1:

**Pull request merged**